src/Arachne.cc

```diff
@@ -1358,18 +1306,27 @@ bool makeExclusiveOnCore(bool forScaleDown) {
             virtualCoreTable[numActiveCores - 1] = core.kernelThreadId;
             break;
         }
+    if (forScaleDown) {
+      coreChangeMutex.lock();
```

Comment:
This seems a little strange: are you doing synchronization here purely because the CorePolicy needs it? If so, then I think you should let the CorePolicy do it for itself. I think I may have changed my mind on this since the last time we discussed this. Let's discuss.